apps/cowswap-frontend/src/common/hooks/useApproveCallback.ts

```diff
@@ -76,10 +76,10 @@ export function useApproveCallback(
       .then((response: TransactionResponse) => {
         addTransaction({
           hash: response.hash,
-          summary: amountToApprove.greaterThan('0') ? `Approve ${token.symbol}` : `Revoke ${token.symbol} approval`,
-          approval: { tokenAddress: token.address, spender, amount: '0x' + amountToApprove.quotient.toString(16) },
+          summary,
+          approval: { tokenAddress: token.address, spender, amount: amountToApproveStr },
         })
         return response
       })
-  }, [chainId, token, tokenContract, amountToApprove, spender, addTransaction])
+  }, [token, tokenContract, spender, addTransaction, summary, amountToApproveStr])
```

Comment:
Fixed [excessive calls](https://github.com/cowprotocol/cowswap/pull/5344#pullrequestreview-2572781798).

It was recalculating the callback because `amountToApprove` is not primitive.